Makefile

```
@@ -40,8 +40,8 @@ ifeq (, $(shell command -v ocb 2>/dev/null))
 	[ "$${machine}" != x86_64 ] || machine=amd64 ;\
 	echo "Installing ocb ($${os}/$${machine}) at $(OTELCOL_BUILDER_DIR)";\
 	mkdir -p $(OTELCOL_BUILDER_DIR) ;\
-	curl -sfLo $(OTELCOL_BUILDER) "https://github.com/open-telemetry/opentelemetry-collector/releases/download/cmd%2Fbuilder%2Fv$(OTELCOL_BUILDER_VERSION)/ocb_$(OTELCOL_BUILDER_VERSION)_$${os}_$${machine}" ;\
-	chmod +x $(OTELCOL_BUILDER) ;\
+	go install go.opentelemetry.io/collector/cmd/builder@v$(OTELCOL_BUILDER_VERSION) ;\
+	mv $$(go env GOPATH)/bin/builder $(OTELCOL_BUILDER) ;\
```

```suggestion
	CGO_ENABLED=0 go build -trimpath -ldflags="-s -w" go.opentelemetry.io/collector/cmd/builder@v$(OTELCOL_BUILDER_VERSION) -o $(OTELCOL_BUILDER) ;\
```
If possible, I wouldn't mind building the module directly:
* Lets us replicate the options used in the goreleaser invocation
* Not a huge deal, but keeps things inside the repo

---

no issues with that approach but I'm going to build and test it in my forked repo for a sanity check

---

seems the only way to use go build with a specific version would be to check out the tagged code directly from the opentelemetry-collector repository and then run the build command on the checked out code.

---

The following code works locally, keeping the flags etc.:
`CGO_ENABLED=0 go install -trimpath -ldflags="-s -w" go.opentelemetry.io/collector/cmd/builder@v$(OTELCOL_BUILDER_VERSION) ;\`
`mv $$(go env GOPATH)/bin/builder $(OTELCOL_BUILDER) ;\`

---

Thanks for investigating that. It looks like you're right, `go install` is the right way to go here.